bblfsh/compat.py

```diff
@@ -0,0 +1,335 @@
+from bblfsh.aliases import (
+    ParseRequest, ParseResponse, DriverStub, ProtocolServiceStub,
+    VersionRequest, SupportedLanguagesRequest, ModeType,
+    Mode, VersionResponse, DESCRIPTOR
+)
+from bblfsh.pyuast import uast, iterator as native_iterator
+from bblfsh.tree_order import TreeOrder
+
+print("Warning: using deprecated bblfsh v1 compatibility layer.", file=sys.stderr)
+
+
+class WrongTypeException(Exception):
+    """
+    This exception is raised when the API receives an unexpected type
+    """
+    pass
+
+
+class CompatParseResponse:
+    """
+    This class emulates the API of the old ParseResponse object.
+    """
+    def __init__(self, ctx: ResultContext, filename: str = "") -> None:
+        self._res_context = ctx
+        self._filename = filename
+
+    @property
+    def uast(self) -> Node:
+        """
+        Returns the root Node.
+        """
+        return self._res_context.uast
+
+    @property
+    def ast(self) -> Node:
+        """
+        Returns the root Node. This is provided for compatibility, but
+        since the type of result is now expecified using CompatBblfshClient.parse
+        or parse_native, it'll return the same as uast().
+        """
+        return self._res_context.ast
+
+    @property
+    def ctx(self) -> ResultContext:
+        """
+        Returns the ResultContext of the response.
+        """
+        return self._res_context
+
+    @property
+    def elapsed(self) -> int:
+        """
+        Provided for compatibility, but since the new API's ParseResponse doesn't
+        provide an elapsed time it'll always return -1.
+        """
+        # FIXME(juanjux): check if the caller can get this, or measure it ourselves.
+        return -1
+
+    @property
+    def language(self) -> str:
+        """
+        Returns the language used for the request.
+        """
+        return self._res_context.language
+
+    @property
+    def filename(self) -> str:
+        """
+        Returns the filename used for the request.
+        """
+        return self._filename
+
+    @property
+    def DESCRIPTOR(self) -> Any:
+        """
+        Returns the gRPC context descriptor.
+        """
+        return self._res_context.ctx.DESCRIPTOR
+
+    @property
+    def errors(selfs) -> List:
+        """
+        Provided for compatibility. Since the new API will raise exceptions on errors,
+        this just returns and empty array.
+        """
+        # ParseResponse would have raised an exception on errors
+        return []
+
+
+class CompatBblfshClient:
+    """
+    This emulates the methods and properties of the old BblfshClient.
+    """
+    def __init__(self, endpoint: Union[str, grpc.Channel]) -> None:
+        """
+        Connects to the specified grpc endpoint which can be specified either as
+        a grpc Channel object or a connection string (like "0.0.0.0:6432").
+        """
+        self._bblfsh_cli = bcli.BblfshClient(endpoint)
```

Comment:
Probably a nitpick, but from the first glance the name `_bblfsh_cli ` made me think about the command line interface and why is it used everywhere.

Reply:
It's from cli-end, but now I see it could be confussing, I've renamed the alias to newbbl (it's only used internally anyway).